pkg/dynamodb/dynamo_api.go

```diff
@@ -0,0 +1,419 @@
+		return conditionBuilder, nil
+	}
+	switch sortKey.ConditionName {
+	case Equal:
+		conditionBuilder = conditionBuilder.And(name(sortKeyName).Equal(value(sortKey.Value)))
+	case BeginsWith:
+		conditionBuilder = conditionBuilder.And(name(sortKeyName).BeginsWith(fmt.Sprintf("%v", sortKey.Value)))
+	default:
+		return expression.ConditionBuilder{}, errors.New(InvalidOperationErrorMessage)
+	}
+
+	return conditionBuilder, nil
+}
+
+func setProjectionBuilder(projectionKeys []string) expression.ProjectionBuilder {
+	var projectionBuilder expression.ProjectionBuilder
+	if len(projectionKeys) > 0 {
+		projectionBuilder = expression.NamesList(name(projectionKeys[0]))
+		if len(projectionKeys) > 1 {
+			for _, val := range projectionKeys[1:] {
+				projectionBuilder = projectionBuilder.AddNames(name(val))
+			}
+		}
+	}
+	return projectionBuilder
+}
+
+func (dynamoClient *DynamoClient) BatchGetItem(ctx context.Context, inputRequest BatchGetItemParams, outputItem interface{}) (interface{}, error) {
+	rows, err := convertKeysToAttributeValues(inputRequest.KeyCondition)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var projections *string
+	var names map[string]string
+
+	if len(inputRequest.ProjectionKeys) > 0 {
+		projectionBuilder := setProjectionBuilder(inputRequest.ProjectionKeys)
+		expr := expression.NewBuilder()
+		expr = expr.WithProjection(projectionBuilder)
+		build, buildErr := expr.Build()
+		if buildErr != nil {
+			return nil, buildErr
+		}
+
+		projections = build.Projection()
+		names = build.Names()
+	}
+
+	input := &dynamodb.BatchGetItemInput{
+		RequestItems: map[string]types.KeysAndAttributes{
+			dynamoClient.Table: {
+				Keys:                     rows,
+				ProjectionExpression:     projections,
+				ExpressionAttributeNames: names,
+				ConsistentRead:           &inputRequest.ConsistentRead,
+			},
+		},
+	}
+
+	result, err := dynamoClient.batchGetItem(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result.UnprocessedKeys) > 0 {
+		return nil, fmt.Errorf("unable to process keys for batch get item, unprocessed keys: %+v", result.UnprocessedKeys)
+	}
+
+	err = attributevalue.UnmarshalListOfMaps(result.Responses[dynamoClient.Table], outputItem)
+	if err != nil {
+		return nil, err
+	}
+	return result.Responses[dynamoClient.Table], err
+}
+
+func (dynamoClient *DynamoClient) BatchPutItem(ctx context.Context, inputItem []interface{}) ([]interface{}, error) {
+	rows, err := convertDataToWriteRequest(inputItem)
+
+	if err != nil {
+		return nil, err
+	}
+
+	input := &dynamodb.BatchWriteItemInput{
+		RequestItems: map[string][]types.WriteRequest{
+			dynamoClient.Table: rows,
+		},
+	}
+
+	result, err := dynamoClient.batchWriteItem(ctx, input)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result.UnprocessedItems) > 0 {
+		unprocessedRows, err := convertWriteRequestItemToInterface(input.RequestItems[dynamoClient.Table])
+		if err != nil {
+			return nil, err
+		}
+		return unprocessedRows, nil
```

Comment:
**Suggestion:** In the `BatchPutItem` method, consider returning the unprocessed items directly if they exist, instead of converting them to an interface, to simplify the error handling logic. [enhancement, importance: 7]
```suggestion
 if len(result.UnprocessedItems) > 0 {
     return result.UnprocessedItems[dynamoClient.Table], nil
 }
```